trio/_core/_run.py

```diff
@@ -116,6 +119,28 @@ class IdlePrimedTypes(enum.Enum):
 ################################################################
 
 
+def collapse_exception_group(excgroup):
+    """Recursively collapse any single-exception groups into that single contained
+    exception.
+
+    """
+    exceptions = list(excgroup.exceptions)
+    modified = False
+    for i, exc in enumerate(exceptions):
+        if isinstance(exc, BaseExceptionGroup):
```

Comment:
I think this should probably only modify trio MultiErrors? It's plausible that someone could raise an exception group from a different library, and we shouldn't mess with it (unlike ours, it might carry important data in its message or other attributes).